docs/upgrading.md

```diff
@@ -0,0 +1,73 @@
+# Upgrading the Agent
+
+This document provides user guidance around agent upgrades and versioning.
+Like every other piece of software included in your project, the agent 
+should be upgraded frequently enough to pick up ongoing enhancements 
+while allowing room for testing. 
+
+## Watching For New Releases
+
+How can you find out about new releases? The best way is to "Watch" this repo
+from the "Watch" button in the upper-right corner of the main repository 
+page. If you don't want to be notified for every pull request and issue, you can 
+choose "Custom" under "Watch" and select just the "Releases" checkbox.
+
+Another approach is to just check in regularly with the GitHub 
+[releases page](https://github.com/signalfx/splunk-otel-java/releases) for this 
+project. 
+
+## When Does Splunk Release?
+
+Because we are a downstream distribution of [OpenTelemetry Java Instrumentation](https://github.com/open-telemetry/opentelemetry-java-instrumentation),
+we generally release within a few working days of every upstream release. We also release when we 
+have new features or enhancements that we want to make available.
+
+We contribute to upstream OpenTelemetry and are typically aware of the release cycle
+over there. As a result, depending on the timing of certain patch releases, we may sometimes
+skip a release to reduce the "release thrash" from our distribution.
+
+## When To Upgrade
+
+In general, we recommend that you upgrade to each new version shortly after it is released. Ideally,
+you should upgrade within a few weeks of a release. Given that OpenTelemetry is a very active 
```

Comment:
I'm not sure what the last sentence means. OpenTelemetry is an active community. Why does this make frequent upgrades important?

Reply:
If you're at version N and you upgrade to N+1, the number of changes in your upgrade path is relatively small. If you wait a year and now need to upgrade from N to N+50, you're taking on a lot more changes all at once and that carries higher risk.